### PR TITLE
wesnoth: update to 1.18.0.

### PR DESCRIPTION
**Pull request by iFoundSilentHouse**
- I tested the changes in this PR: **YES**
Checked out new compaign.

- I built this PR locally for my native architecture, (x86_64-glibc)

Patch for boost is no longer needed as it was commited to 1.17 version.
Added libcurl as changelog says wesnoth now needs it for compiling.

### CODE REVIEW

**Comment by iFoundSilentHouse** [comment]
For x86_64 build is fine locally (I've checked it multiple times). Though for musl some extra patching needs to be done. Working on it.

**Comment by iFoundSilentHouse** [comment]
2024-03-26-145847_1920x1080_scrot
2024-03-26-155002_1920x1080_scrot
2024-03-26-161016_1920x1080_scrot

**Comment by iFoundSilentHouse** [comment]
Added pull request on wesnoth to fix this bug: https://github.com/wesnoth/wesnoth/pull/8628
Patch from https://patch-diff.githubusercontent.com/raw/wesnoth/wesnoth/pull/8628.patch doesn't work, though patch made locally works fine.

File: srcpkgs/wesnoth/template [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="GPL-2.0-or-later"
 homepage="https://wesnoth.org"
+changelog="https://github.com/wesnoth/wesnoth/blob/master/changelog.md"
 distfiles="${SOURCEFORGE_SITE}/${pkgname}/${pkgname}-${version}.tar.bz2"
-checksum=a60521b1ee1c2ddc7dfde89a038b893a4b43337085bd1478f4bc92dab7da7fc7
+checksum=2cdc75b97c3437e6490f1982c7938e87d5dcbe29ca8aed32a8d13b25e06939c0
 replaces="wesnoth-data>=0"
+case "$XBPS_TARGET_MACHINE" in
+	x86_64*|i686*) make_check=no ;; # Cannot reproduce fail locally
```
Comment (classabbyamp) [inline srcpkgs/wesnoth/template:21]: if it's only in ci, it should be `ci-skip`

**Comment by iFoundSilentHouse** [comment]
```
Start 6: SDL2_WEBP_SUPPORT
6/7 Test #6: SDL2_WEBP_SUPPORT ................***Failed    0.00 sec
```
I don't know what to do. Old SDL2_image version doesn't seem to be a thing

**Comment by iFoundSilentHouse** [comment]
I assume that the only way to build wesnoth with webp support (which is neccesary for testing environment) using dynamic linking is to add patch directly to CMakeLists.txt. Other ways of using webp require static linking

**Comment by iFoundSilentHouse** [comment]
Changes: 
* SDL2_image: for webp support added patch to SDL2_image as there's no command that worked for build using dynamic linking.
* wesnoth: 1) Added patch that allows musl to build; this code doesn't change functionality 2) Disabled audio test - obviously, virtual environment doesn't have sound card.
Everything works. @classabbyamp

**Comment by iFoundSilentHouse** [comment]
> > * SDL2_image: for webp support added patch to SDL2_image as there's no command that worked for build using dynamic linking.
> 
> Instead of adding a patch, add `configure_args="-DSDL2IMAGE_WEBP=ON"` to SDL2_image's template.
> 
> If done correctly, this should be part of the configure step's output for SDL2_image:
> 
> ```
> -- SDL2_image: Using system libwebp
> -- Found webp: /usr/lib/libwebp.so  
> -- Dynamic libwebpdemux: libwebpdemux.so.2
> -- Dynamic libwebp: libwebp.so.7
> ```

I tried it. Didn't work. Maybe I was too tired bruteforcing the answer and did something wrong. I'll try it againg

**Comment by iFoundSilentHouse** [comment]
> > I tried it. Didn't work. Maybe I was too tired bruteforcing the answer and did something wrong. I'll try it againg
> 
> You might need to `./xbps-src clean SDL2_image` first for some reason.
> 
> Thanks for working on the update, by the way. I do want to play Wesnoth 1.18 at some point.

Thanks for your reviews a lot! :)

File: srcpkgs/wesnoth/template [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
 makedepends="dbus-devel libgomp-devel fribidi-devel SDL2_image-devel
  SDL2_net-devel SDL2_ttf-devel SDL2_mixer-devel pango-devel libvorbis-devel
- zlib-devel bzip2-devel boost-devel openssl-devel SDL2-devel"
+ zlib-devel bzip2-devel boost-devel openssl-devel SDL2-devel libcurl-devel"
 depends="desktop-file-utils"
 short_desc="Turn-based fantasy strategy game"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="GPL-2.0-or-later"
 homepage="https://wesnoth.org"
+changelog="https://github.com/wesnoth/wesnoth/blob/master/changelog.md"
```
[the remark this answers is not in the document]
Comment (classabbyamp) [reply srcpkgs/wesnoth/template:16]: no, we want the changelog to match the development branch if it's in the git repo

[the remark this answers is not in the document]
Comment (classabbyamp) [reply srcpkgs/wesnoth/template:16]: the change that should be made is switching from the "blob" url to the "raw" url

[the remark this answers is not in the document]
Comment (iFoundSilentHouse) [reply srcpkgs/wesnoth/template]: > the change that should be made is switching from the "blob" url to the "raw" url

Changed

**Comment by iFoundSilentHouse** [comment]
Added wesnoth developers' patch to fix musl instead of mine: https://github.com/wesnoth/wesnoth/pull/8649 
This one is more accurate

File: srcpkgs/wesnoth/patches/001-disable-audio-test.patch
Comment (classabbyamp) [inline srcpkgs/wesnoth/patches/001-disable-audio-test.patch:1]: why is this patch needed? add some context at the beginning of the patch explaining please

Comment (iFoundSilentHouse) [reply srcpkgs/wesnoth/patches/001-disable-audio-test.patch:1]: Thank you for your attention. done

**Pull request merged**